databuilder/query_engines/base_sql.py

```diff
@@ -221,6 +221,14 @@ def get_sql_subtract(self, node):
     def get_sql_multiply(self, node):
         return operators.mul(self.get_expr(node.lhs), self.get_expr(node.rhs))
 
+    @get_sql.register(Function.TrueDivide)
+    def get_sql_truedivide(self, node):
+        return operators.truediv(self.get_expr(node.lhs), self.get_expr(node.rhs))
```

Comment:
SQL doesn't have distinct truediv/floordiv operators. Instead it says, if both arguments are ints we perform floor division, otherwise we perform true division. So to get the behaviour we want we have to cast types appropriately. Something like this:
```py
lhs = self.get_expr(node.lhs)
rhs = self.get_expr(node.rhs)
# If both arguments are ints, cast one to a float to ensure we get TrueDiv behaviour
if isinstance(lhs.type, sqlalchemy_types.Integer) and isinstance(rhs.type, sqlalchemy_types.Integer):
    lhs = sqlalchemy.cast(lhs, sqlalchemy.Float)
return operators.div(lhs, rhs)
```